### DISCUSSION TITLE
Problem in creating surfaces

### DISCUSSION THREAD

**Discussion by KodalAnwesha** [gempy-project/gempy#947] [Q&A]
Hello,

I was trying to create surfaces from the coordinates and orientations files. In my case the X, Y, Z coordinates are very close to each other. 
The problem I am facing is, the surfaces are not getting generated, the orientation direction is also not showing.

In the 2D plot nothing is visible properly and the 3D plot is showing some invalid surface.

Please help me out with this.

Thanks!
![Screenshot 2024-10-22 120931](https://github.com/user-attachments/assets/05c35cef-455c-4888-ac30-a9592f867772)
![Screenshot 2024-10-22 120949](https://github.com/user-attachments/assets/7d9a359d-8715-4abf-b5cc-67a71667dee2)
![Screenshot 2024-10-22 120848](https://github.com/user-attachments/assets/0253d45d-4464-497d-942a-d288e49e7681)

**Comment by javoha** [comment]
Hi,
from your images it is really hard to see what this is and what you try to do. What is the resolution of your model? And maybe you can increase the vertical exaggeration in your plots for better visualization.
Cheers,
Jan

**Comment by KodalAnwesha** [comment]
Hello,

I am providing the used dataset, can you please try plotting it once. Maybe
you can get my issue then.

Thank you!

**Comment by javoha** [comment]
Hi, 
where are you providing the dataset? And have you tried my suggestions: Increase resolution and vertical exaggeration in z for the plots?

**Reply by KodalAnwesha** [reply]
Hello,
Have you tried generating the model?

**Comment by KodalAnwesha** [comment]
Hello,

I had already attached the CSV files with this email.
And yes, I tried your suggestion of increasing the vertical exaggeration,
but still it's showing some absurd model.

Thank you
Anwesha

**Comment by flohorovicic** [comment]
I also can't see an attached file here in the discussion. Maybe better to try in an issue, if it doesn't work here.

A quick point, looking at your model as far as we can see in the plots above: there are only a few points in the middle, very close by. Which could at least explain the shape.

Also, did you try setting an anisotropy before computing?

Could also help:

```
# Set global anisotropy before model computation
geo_model.update_transform(gp.data.GlobalAnisotropy.NONE)

# Compute the model
gp.compute_model(geo_model, engine_config=gp.data.GemPyEngineConfig())
```

**Comment by KodalAnwesha** [comment]
Thank you. I figured this out.

**Reply by flohorovicic** [reply]
Can you tell us a bit more about it? How did you solve it? Certainly interesting also for others :-)
